File: passenger/src/main/java/com/modsen/passenger/repository/PassengerRepository.java
package com.modsen.passenger.repository;

import com.modsen.passenger.model.Passenger;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface PassengerRepository extends JpaRepository<Passenger, Integer> {

    Optional<Passenger> findByEmail(String email);

    Optional<Passenger> findByPhoneNumber(String phoneNumber);
}

File: passenger/src/main/kotlin/com/modsen/ktpassenger/repository/PassengerRepository.kt
package com.modsen.ktpassenger.repository

import com.modsen.ktpassenger.model.Passenger
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface PassengerRepository : JpaRepository<Passenger, Int> {
    fun findByEmail(email: String): Optional<Passenger>

    fun findByPhoneNumber(phoneNumber: String): Optional<Passenger>
}